src/Controller/MediaAdminController.php

```diff
@@ -72,7 +72,7 @@ public function listAction(Request $request): Response
         $datagrid->setValue('context', null, $context);
 
         $rootCategory = null;
-        if ($this->has('sonata.media.manager.category') && $this->has('sonata.media.manager.context')) {
+        if ($this->has('sonata.media.manager.category')) {
```

Comment:
adding this checks makes phpstan not happy...

```
 ------ ----------------------------------------------------------------------------------------------------------------------------------------
  Line   src/Controller/MediaAdminController.php
 ------ ----------------------------------------------------------------------------------------------------------------------------------------
  80     Call to method Symfony\Bundle\FrameworkBundle\Controller\AbstractController::has() with 'sonata.media…' will always evaluate to false.
 ------ ----------------------------------------------------------------------------------------------------------------------------------------
```

Super strange, if I try to split the if like:

```
if ($this->has()) {
}
if ($this->has() {
}
```

it works, but having the condition in the same if or nested it fails.

Reply:
Both has should be kept since we're using both get.

I'll take a look, but maybe it will be better to ignore the error and report an issue to phpstan.

Reply:
I think is because of the phpstan-symfony plugin, if I disable it, the error goes away.

Reply:
Just added an issue: https://github.com/phpstan/phpstan-symfony/issues/178

Reply:
Seems like there is something about in the doc Seems related to https://github.com/phpstan/phpstan-symfony#constant-hassers

Reply:
I tried to set it to false, didn't worked, will try later again.

Reply:
Isn't `sonata.classification.manager.context` instead ?

https://github.com/sonata-project/SonataClassificationBundle/blob/master/src/Resources/config/orm.xml#L23

Reply:
Nope, it is sonata.media.manager.context: https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Resources/config/category.php

those alias are created because on media-bundle you can decide (even if you have classification-bundle installed) that you don't want to use it.

Maybe it is a bit weird that works this way but for now it is how it is implemented (maybe we can deprecate that later).

Reply:
I ignored the error for now, the code is correct.

Reply:
It's fixed in the last version of phpstan